docs: route groups (#572)
posva/unplugin-vue-router: docs/guide/file-based-routing.md

All generated routes that have a `component` property will have a `name` property. This avoid accidentally directing your users to a parent route. By default, names are generated using the file path, but you can override this behavior by passing a custom `getRouteName()` function. You will get TypeScript validation almost everywhere, so changing this should be easy.

> ### Route groups

Reviewer: ```suggestion
## Route groups
```

Sometimes, it helps to organize your file structure in a way that doesn't change the URL of your routes. Route groups let you organize your routes logically, in a way that makes sense to you, without affecting the actual URLs. For example, if you have several routes that share the same layout, you can group them together using route groups.

```text
src/pages/
├── (admin)/
│   ├── dashboard.vue
│   ├── settings.vue
├── (user)/
│   ├── profile.vue

> │   ├── orders.vue

Reviewer: Use the correct characters like in the other examples. You can also use the `tree` command (brew install tree)` to print these:

```
└── vim                                                                                                                                         
    ├── colors                                                                                                                                  
    │   ├── distinguished.vim                                                                                                                   
    │   ├── lucius.vim                                                                                                                          
    │   ├── molokai.vim                                                                                                                         
    │   └── neverness.vim                                                                                                                       
    ├── ftplugin                                                                                                                                
    │   ├── c.vim                                                                                                                               
    │   ├── cpp.vim                                                                                                                             
    │   └── java.vim                                                                                                                            
    └── spell                                                                                                                                   
        ├── en.utf-8.add                                                                                                                        
        ├── en.utf-8.add.spl                                                                                                                    
        ├── es.utf-8.add                                                                                                                        
        ├── es.utf-8.add.spl                                                                                                                    
        ├── es.utf-8.spl                                                                                                                        
        ├── es.utf-8.sug                                                                                                                        
        ├── fr.utf-8.spl                                                                                                                        
        └── fr.utf-8.sug   
```

> ```

Reviewer: After showing the folder structure, it's important to show the resulting routes as a list and what component they show

Author: @posva I updated this again for better clarity:  https://github.com/posva/unplugin-vue-router/pull/572/commits/ed3971f937e1ccaa134500e61d4b4de2e6139466

> You can also use file route groups for better organization. For example:

Reviewer: ```suggestion
You can also use route groups in page components. This is equivalent to name the page component `index.vue`:
```

All generated routes that have a `component` property will have a `name` property. This avoid accidentally directing your users to a parent route. By default, names are generated using the file path, but you can override this behavior by passing a custom `getRouteName()` function. You will get TypeScript validation almost everywhere, so changing this should be easy.

## Route groups

Sometimes, it helps to organize your file structure in a way that doesn't change the URL of your routes. Route groups let you organize your routes logically, in a way that makes sense to you, without affecting the actual URLs. For example, if you have several routes that share the same layout, you can group them together using route groups. Consider the following file structure:

```text

> src/pages/
> ├── (admin)/
> │   ├── dashboard.vue
> │   └── settings.vue
> ├── (user)/
> │   ├── profile.vue
> │   └── order.vue

Reviewer: ```suggestion
src/pages/
├── (admin)/
│   ├── dashboard.vue
│   └── settings.vue
└── (user)/
    ├── profile.vue
    └── order.vue
```

```

Resulting URLs:
```text

> - `/dashboard` -> renders `dashboard.vue`

Reviewer: let's use the full path here for clarity
```suggestion
- `/dashboard` -> renders `src/pages/(admin)/dashboard.vue`
```

- `/settings` -> renders `src/pages/(admin)/settings.vue`
- `/profile` -> renders `src/pages/(user)/profile.vue`
- `/order` -> renders `src/pages/(user)/order.vue`
```

> This approach allows you to organize your files for better maintainability without changing the structure of your application's routes.

Reviewer: ```suggestion

This approach allows you to organize your files for better maintainability without changing the structure of your application's routes.
```

You can also use route groups in page components. This is equivalent to name the page component `index.vue`:

```text
src/pages/

> ├── admin/
> │   ├── (dashboard).vue // Becomes index.vue of admin route
> │   └── settings.vue

Reviewer: ```suggestion
└─── admin/
    ├── (dashboard).vue // Becomes index.vue of admin route
    └── settings.vue
```
